src/Symfony/Component/Ldap/LdapClient.php

```diff
@@ -0,0 +1,220 @@
+<?php
+
+namespace Symfony\Component\Ldap;
+
+use Symfony\Component\Ldap\Exception\ConnectionException;
+use Symfony\Component\Ldap\Exception\LdapException;
+
+/**
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ * @author Francis Besset <francis.besset@gmail.com>
+ * @author Charles Sarrazin <charles@sarraz.in>
+ */
+class LdapClient implements LdapClientInterface
+{
+    private $host;
+    private $port;
+    private $version;
+    private $useSsl;
+    private $useStartTls;
+    private $optReferrals;
+    private $connection;
+    private $charmaps;
+
+    /**
+     * Constructor.
+     *
+     * @param string $host
+     * @param int    $port
+     * @param int    $version
+     * @param bool   $useSsl
+     * @param bool   $useStartTls
+     * @param bool   $optReferrals
+     */
+    public function __construct($host = null, $port = 389, $version = 3, $useSsl = false, $useStartTls = false, $optReferrals = false)
+    {
+        if (!extension_loaded('ldap')) {
+            throw new LdapException('The ldap module is needed.');
+        }
+
+        $this->host = $host;
+        $this->port = $port;
+        $this->version = $version;
+        $this->useSsl = (bool) $useSsl;
+        $this->useStartTls = (bool) $useStartTls;
+        $this->optReferrals = (bool) $optReferrals;
+    }
+
+    public function __destruct()
+    {
+        $this->disconnect();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function bind($dn = null, $password = null)
+    {
+        if (!$this->connection) {
+            $this->connect();
+        }
+
+        if (false === @ldap_bind($this->connection, $dn, $password)) {
+            throw new ConnectionException(ldap_error($this->connection));
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function find($dn, $query, $filter = '*')
+    {
+        if (!is_array($filter)) {
+            $filter = array($filter);
+        }
+
+        $search = ldap_search($this->connection, $dn, $query, $filter);
+        $infos = ldap_get_entries($this->connection, $search);
+
+        if (0 === $infos['count']) {
+            return;
+        }
+
+        return $infos;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function escape($subject, $ignore = '', $flags = 0)
+    {
+        if (function_exists('ldap_escape')) {
+            return ldap_escape($subject, $ignore, $flags);
+        }
+
+        return $this->doEscape($subject, $ignore, $flags);
+    }
+
+    private function connect()
```

Comment:
I'm repeating myself, but the connection logic will be moved out from the client. So indeed, you will be able to configure all connection options (as well as connection handling) the Connection class.